lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -259,205 +272,224 @@ def run(self, **kwargs):
+                self.update_pillar_and_conf('user', user)
+                self.update_pillar_and_conf('password', password)
+            elif self.enclosure_id is not None and setup_type == "HW":
+                # Store user and password only after validation
+                # Skip the validation if enclosure id was fetched
+                #  using the same credentials
+                if not cred_validation:
+                    # Read ip & port from Pillar and validate by logging
+                    # in to enclosure with user, passwd, ip and port
+                    self.logger.debug("Validating the user and password provided")
+                    host_in_pillar = get_pillar_data(
+                        f"storage/{enc_num}/controller/primary/ip")
+                    port_in_pillar = get_pillar_data(
+                        f"storage/{enc_num}/controller/primary/port")
+                    if not host_in_pillar or not port_in_pillar:
+                        self.logger.error(
+                            f"Could not read controller ip and secret from pillar")
+                        raise RuntimeError("Could not validate user and password")
+                    valid_connection_check = EnclosureInfo(
+                                                host_in_pillar,
+                                                user,
+                                                password,
+                                                port_in_pillar
+                                            ).connection_status()
+                    if not valid_connection_check:
+                        self.logger.error(
+                            f"Could not establish connection with"
+                            " controller with provided credentials"
+                        )
+                        raise ValueError("Invalid credentials provided")
                 self.update_pillar_and_conf('user', user)
-
-                # store password
                 self.update_pillar_and_conf('password', password)
+            else:
+                self.logger.error(f"Enclosure ID is not set\n"
+                    "Run following command to set the enclosure id:"
+                    "cortx_setup storage config --user <user>"
+                    " --password <passwd> --ip <ip> --port <port>")
+                raise RuntimeError(
+                    "Cannot set mode, ip and port without enclosure id"
+                )
 
-                # store ip and port as primary
-                self.update_pillar_and_conf('ip', ip)
-                self.update_pillar_and_conf('port', port)
+        if ip is not None or port is not None:
+            if self.mode is None:
+                self.logger.exception(
+                    f"mode is missing, please provide --mode argument"
+                )
+                raise RuntimeError("Incomplete arguments provided")
             else:
```

Comment:
Why do  we need else block, please see if this can be handled in outside ese block itself

Reply:
It does nothing, just for readability and to understand the code better.